lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -960,3 +1026,72 @@ def _clear_errors(self, *error_labels: str):
             error = self.charm.peers_data.get(Scope.APP, error_label, "")
             self.charm.status.clear(error, app=True)
             self.charm.peers_data.delete(Scope.APP, error_label)
+
+
+def rel_data_from_redacted_str(
+    charm: "OpenSearchBaseCharm", redacted_dict_str: str
+) -> PeerClusterRelData | None:
+    """Construct the peer cluster rel data from the secret data."""
+    content = json.loads(redacted_dict_str)
+    credentials = content["credentials"]
+
+    secrets = charm.secrets
```

Comment:
Should be moved to the initialization of the class.

Reply:
This is outside the class because it is also used in the `opensearch_peer_clusters` module